numpy/_core/src/umath/loops_hyperbolic.dispatch.cpp.src

```diff
@@ -73,7 +77,145 @@
  *         achieve wider than target precision.
  *
  */
-#if NPY_SIMD_F64
+
+namespace hn = hwy::HWY_NAMESPACE;
+
+const hn::ScalableTag<float> f32;
+const hn::ScalableTag<double> f64;
+const hn::ScalableTag<int32_t> s32;
+const hn::ScalableTag<int64_t> s64;
+const hn::ScalableTag<uint32_t> u32;
+const hn::ScalableTag<uint64_t> u64;
+using vec_f32 = hn::Vec<decltype(f32)>;
+using vec_f64 = hn::Vec<decltype(f64)>;
+using vec_s32 = hn::Vec<decltype(s32)>;
+using vec_s64 = hn::Vec<decltype(s64)>;
+using vec_u32 = hn::Vec<decltype(u32)>;
+using vec_u64 = hn::Vec<decltype(u64)>;
+using opmask_t = hn::Mask<decltype(f32)>;
+
+struct hwy_f32x2 {
+    vec_f32 val[2];
+};
+
+HWY_ATTR NPY_FINLINE hwy_f32x2 zip_f32(vec_f32 a, vec_f32 b){
+    hwy_f32x2 res;
+    res.val[0] = hn::InterleaveLower(f32, a, b);
+    res.val[1] = hn::InterleaveUpper(f32, a, b);
+    return res;
+}
+
+HWY_ATTR NPY_FINLINE vec_f64 lut_16_f64(const double * lut, vec_u64 idx){
+    if constexpr(hn::Lanes(f64) == 8){
+        const vec_f64 lut0 = hn::Load(f64, lut);
+        const vec_f64 lut1 = hn::Load(f64, lut + 8);
+        return hn::TwoTablesLookupLanes(f64, lut0, lut1, hn::IndicesFromVec(f64, idx));
```

Comment:
It might actually be worthwhile to implement an AVX2 version with 4 table lookups and blending, because gather is quite slow again.